nextcord/automod.py

```diff
@@ -0,0 +1,333 @@
+Copyright (c) 2022-present tag-epic
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, overload
+
+from .enums import ActionType, EventType, KeywordPresetType, TriggerType, try_enum
+from .mixins import Hashable
+
+if TYPE_CHECKING:
+    from datetime import datetime
+    from typing import List, Optional
+
+    from .abc import GuildChannel
+    from .errors import InvalidArgument
+    from .guild import Guild
+    from .role import Role
+    from .state import ConnectionState
+    from .types.automod import (
+        ActionMetadata as ActionMetadataPayload,
+        AutoModerationAction as AutoModerationActionPayload,
+        AutoModerationRule as AutoModerationRulePayload,
+        TriggerMetadata as TriggerMetadataPayload,
+    )
+    from .utils import MISSING, snowflake_time
+
+__all__ = ("AutoModerationRule", "AutoModerationAction")
+
+
+class AutoModerationAction:
+    """
+    Represent an auto moderation action.
+
+    Attributes
+    -----------
+    type: :class:`ActionType`
+        The action type of this action.
+    notify_channel_id: Optional[:class:`int`]
+        The ID of the channel that this action will notify in.
+    timeout_seconds: Optional[:class:`int`]
+        The number of seconds this rule should timeout when someone breaks."""
+
+    def __init__(self, data: AutoModerationActionPayload):
+        self.type: ActionType = try_enum(ActionType, data["type"])
+        self.notify_channel_id: Optional[int] = None
+        self.timeout_seconds: Optional[int] = None
+        if data.get("metadata"):
+            self._unpack_metadata(data["metadata"])  # type: ignore
+
+    def __repr__(self):
+        attrs = (
+            ("type", self.type),
+            ("notify_channel_id", self.notify_channel_id),
+            ("timeout_seconds", self.timeout_seconds),
+        )
+        inner = " ".join("%s=%r" % t for t in attrs)
+        return f"<AutoModerationAction {inner}>"
```

Comment:
Can't this just be hardcoded?

Reply:
wdym?

Reply:
Why is the `.join` required for 2 parameters

Reply:
consistent with other parts of the lib

Reply:
There are some classes that just use a plain f string if there's small enough amounts of attributes